modules/installation-performing-disconnected-mirror-without-registry.adoc

```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_restricted_networks/installing-restricted-networks-preparations.adoc
+
+[id="installation-performing-disconnected-mirror-without-registry"]
+= Performing a mirror to disk for use in disconnected environments with a non-production mirror registry
+
+If a production mirror registry is not available, you can configure a simple mirror registry by using the disconnected procedure to serve container images that you downloaded to disk.
+
+.Procedure
+
+. Determine the IP of your host within the restricted network as `<private_ip>`.
+
+. On a local host, copy the ICS/ICSP:
+.. Make the mirror directory and change to it:
++
+----
+$ mkdir <mirror_dir> ; cd <mirror_dir>
+----
+
+.. Mirror the images:
++
+----
+$ oc adm release mirror <product_version> --to file://openshift/release
+----
+
+. From within the restricted network, start an image mirror server on port 5000 on all interfaces on the host:
+.. Change to the mirror directory:
++
+----
+$ cd MIRROR_DIR
+----
+
+.. Serve the images for the installation program to use:
++
+----
+$ oc image serve
+----
+
+This registry does not perform authentication and does not require TLS in order to guarantee integrity of the provided images.
```

Comment:
From my understanding, here is a huge gap to get this process working, I never testes this before.
1. Here is my trial steps:
`
$ oc adm release mirror <product_version> --to file://test
<--snip-->
Success
Update image:  test:4.3.0-0.nightly-2020-06-07-045523
Mirror prefix: file://test

To upload local images to a registry, run:

    oc image mirror --from-dir=/home/installer1/mnt/2020-06-08-07-57-52/jialiu432_2020-06-08-07-57-52/data file://test:4.3.0-0.nightly-2020-06-07-045523* REGISTRY/REPOSITORY

$ oc image serve
error: --dir must point to a directory: stat : no such file or directory

$ oc image serve --dir=./data --listen=:5001
I0628 06:09:47.819085   21665 serve.go:119] Serving at :5001 ...

$ podman pull upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523 --tls-verify=false --log-level debug
DEBU[0000] Initializing boltdb state at /var/lib/containers/storage/libpod/bolt_state.db 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /var/lib/containers/storage 
DEBU[0000] Using run root /var/run/containers/storage   
DEBU[0000] Using static dir /var/lib/containers/storage/libpod 
DEBU[0000] Using tmp dir /var/run/libpod                
DEBU[0000] Using volume path /var/lib/containers/storage/volumes 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] cached value indicated that overlay is supported 
DEBU[0000] cached value indicated that metacopy is not being used 
DEBU[0000] cached value indicated that native-diff is usable 
DEBU[0000] backingFs=xfs, projectQuotaSupported=false, useNativeDiff=true, usingMetacopy=false 
DEBU[0000] Initializing event backend journald          
INFO[0000] Found CNI network crio-bridge (type=bridge) at /etc/cni/net.d/100-crio-bridge.conf 
INFO[0000] Found CNI network 200-loopback.conf (type=loopback) at /etc/cni/net.d/200-loopback.conf 
INFO[0000] Found CNI network podman (type=bridge) at /etc/cni/net.d/87-podman-bridge.conflist 
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/var/run/containers/storage]upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523" 
Trying to pull upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523...DEBU[0000] reference rewritten from 'upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523' to 'upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523' 
DEBU[0000] Trying to pull "upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523" 
DEBU[0000] Using registries.d directory /etc/containers/registries.d for sigstore configuration 
DEBU[0000]  Using "default-docker" configuration        
DEBU[0000]  No signature storage configuration found for upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523 
DEBU[0000] Looking for TLS certificates and private keys in /etc/docker/certs.d/upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001 
DEBU[0000] GET https://upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/v2/ 
DEBU[0000] Ping https://upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/v2/ err Get https://upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/v2/: http: server gave HTTP response to HTTPS client (&url.Error{Op:"Get", URL:"https://upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/v2/", Err:(*errors.errorString)(0xc420462090)}) 
DEBU[0000] GET http://upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/v2/ 
DEBU[0000] Ping http://upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/v2/ status 200 
DEBU[0000] GET http://upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/v2/file/manifests/4.3.0-0.nightly-2020-06-07-045523 
ERRO[0000] Error pulling image ref //upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523: Error initializing source docker://upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523: Error reading manifest 4.3.0-0.nightly-2020-06-07-045523 in upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file: error parsing HTTP 404 response body: invalid character 'p' after top-level value: "404 page not found\n" 
Failed
ERRO[0000] error pulling image "upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523": unable to pull upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523: unable to pull image: Error initializing source docker://upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file:4.3.0-0.nightly-2020-06-07-045523: Error reading manifest 4.3.0-0.nightly-2020-06-07-045523 in upshift-nointernet.mirror-registry.qe.devcluster.openshift.com:5001/file: error parsing HTTP 404 response body: invalid character 'p' after top-level value: "404 page not found\n" 
`
Even I applied some workaround, still no chance. What is next steps to consume this mirrored release image?

2. In cluster install process. podman always use https by default to pull image, such as: 
[1]: https://github.com/openshift/installer/blob/master/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L13
[2]: https://github.com/openshift/installer/blob/master/data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template#L15
In this section, these steps start a non-insecure registry, that definitely does not work with openshift v4 (all image-pull places reqire HTTPS). Another potential issue this even does not need authentication, how to help user set pull secret in install-config to allow customer to follow the next intsall section to get a successfull install?

Reply:
@wking, what's your opinion on including the non-production mirror steps? Are you ok with making an existing registry a requirement, or is there a way to make this process work?

Reply:
I am find punting as much as we can so we can land the non-contentious bits :)